web-client/src/presenter/actions/updatePartyTypeAction.test.js

```diff
@@ -0,0 +1,190 @@
+    const { state } = await runAction(
+      updatePartyTypeAction,
+      getFixtures({
+        key: 'isSpouseDeceased',
+        value: 'Yes',
+      }),
+    );
+    expect(state.form.partyType).toEqual('Petitioner & Deceased Spouse');
+  });
+
+  it('sets the partyType to "Petitioner & Spouse" when "isSpouseDeceased" is updated to "No"', async () => {
+    const { state } = await runAction(
+      updatePartyTypeAction,
+      getFixtures({
+        key: 'isSpouseDeceased',
+        value: 'No',
+      }),
+    );
+    expect(state.form.partyType).toEqual('Petitioner & Spouse');
+  });
+
+  it('sets the partyType to "Donor" when "otherType" is updated to "Donor"', async () => {
+    const { state } = await runAction(
+      updatePartyTypeAction,
+      getFixtures({
+        key: 'otherType',
+        value: 'Donor',
+      }),
+    );
+    expect(state.form.partyType).toEqual('Donor');
+  });
+
+  it('sets the partyType to "Transferee" when "otherType" is updated to "Transferee"', async () => {
+    const { state } = await runAction(
+      updatePartyTypeAction,
+      getFixtures({
+        key: 'otherType',
+        value: 'Transferee',
+      }),
+    );
+    expect(state.form.partyType).toEqual('Transferee');
+  });
+
+  it('sets the partyType to "Surviving Spouse" when "otherType" is updated to "Deceased Spouse"', async () => {
+    const { state } = await runAction(
+      updatePartyTypeAction,
+      getFixtures({
+        key: 'otherType',
+        value: 'Deceased Spouse',
+      }),
+    );
+    expect(state.form.partyType).toEqual('Surviving Spouse');
+  });
+
+  it('sets the partyType to the props.value passed in when key is "businessType"', async () => {
+    const { state } = await runAction(
+      updatePartyTypeAction,
+      getFixtures({
+        key: 'businessType',
+        value: 'Any Value',
+      }),
+    );
+    expect(state.form.partyType).toEqual('Any Value');
+  });
+
+  it('sets the partyType to the props.value passed in when the key is "estateType" and set form.otherType to "An estate or trust"', async () => {
+    const { state } = await runAction(
+      updatePartyTypeAction,
+      getFixtures({
+        key: 'estateType',
+        value: 'Any Value',
+      }),
+    );
+    expect(state.form.partyType).toEqual('Any Value');
+    expect(state.form.otherType).toEqual('An estate or trust');
+  });
+
+  it('sets the partyType to the props.value passed in when the key is "minorIncompetentType" and set form.otherType to "A minor or incompetent person"', async () => {
+    const { state } = await runAction(
+      updatePartyTypeAction,
+      getFixtures({
+        key: 'minorIncompetentType',
+        value: 'Any Value',
+      }),
+    );
+    expect(state.form.partyType).toEqual('Any Value');
+    expect(state.form.otherType).toEqual('A minor or incompetent person');
+  });
+
+  it('sets the petition.ownershipDisclosureFile and form.businessType when form.filingType is not "A business"', async () => {
```

Comment:
I'm puzzled about this. The description says that the `filingType` is `not "A business"`, but the test is looking for a `filingType` of `Not a Business`. Presumably the description is accurate, but the test is wrong, and there's not actually a `Not a Business` filing type?

Reply:
I'm going to rename the test to this if this makes more sense: "resets the state.petition.ownershipDisclosureFile and state.form.businessTyp when form.filingType is anything other than "A business""

Reply:
Basically, if filingType is some random string, it will set those 2 state properties to undefined.

Reply:
I do think that renamed test helps. I'm tempted to say "nah, this was just me being dense," but I do think it's tricky when verifying that a string is not anything at all other than the one correct value — just about any value you test is confusing. The only solution is a test name that captures that, which I think your suggested test name captures well. Thanks!